### PR TITLE
Preserve data in the browser

### PR DESCRIPTION
**Pull request by MuhmmdUsama**
- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -195,4 +195,43 @@ form.addEventListener('submit', (event) => {
...
+      localStorage.setItem('data', JSON.stringify(storagData));
+    });
+  });
+  const getData = JSON.parse(localStorage.getItem('data'));
+  if (getData) {
+    form.name.value = getData.name;
+    form.email.value = getData.email;
+    form.message.value = getData.message;
+  }
+}
```
Comment (mateo951) [inline script.js]: There is a tiny bug with your code at the moment. Your email and text are being saved and the persistence is working for them but not for the name. As you can see in the following image, the name text is changed to undefined. 

image

Kindly make the necessary changes to solve the issue described 🧐

File: script.js [hunk 2]
```diff
@@ -195,4 +195,43 @@ form.addEventListener('submit', (event) => {
...
+    input.addEventListener('input', () => {
+      const storagData = {
+        fullName: form.name.value,
+        email: form.email.value,
+        message: form.message.value,
+      };
+      localStorage.setItem('data', JSON.stringify(storagData));
+    });
+  });
+  const getData = JSON.parse(localStorage.getItem('data'));
```
Comment (mateo951) [inline script.js:231]: Nice 🔥 The form data for all inputs are stored in a single object, saved to a single local storage registry 💪😎

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi Team, 

👌 **Your project highlights** 😎 
- Linters are working properly ✔
- Nice README file 🤓
- GitHub flow used as intended 🔥

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: script.js (hunk 1 above)
Comment (MuhmmdUsama) [reply script.js]: we did it

**Review by CollinsTatang — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**